test/unit/spec/VerifyCustomFactor_spec.js

```diff
@@ -0,0 +1,268 @@
+          expect(test.form.hasSpinner()).toBe(false);
+        });
+      });
+
+      itp('shows the right auto redirect subtitle', function() {
+        return setup(settings)
+          .then(function(test) {
+            expectSubtitleToBe(test, 'Redirecting to IDP factor...');
+          });
+      });
+
+      itp('shows the right auto redirect subtitle when lastFailedChallengeFactorData is not null', function() {
+        return setup(settings, lastFailedChallengeFactorData).then(function(test) {
+          expectSubtitleToBe(test, 'Clicking below will redirect to verification with IDP factor');
+        });
+      });
+
+      itp('redirects automatically to third party', function() {
+        spyOn(SharedUtil, 'redirect');
+        return setup(settings)
+          .then(function(test) {
+            test.setNextResponse([resChallengeClaimsProvider, resSuccess]);
+            return Expect.waitForSpyCall(SharedUtil.redirect);
+          })
+          .then(function() {
+            expect(SharedUtil.redirect).toHaveBeenCalledWith(
+              'http://rain.okta1.com:1802/sso/idps/idpId?stateToken=testStateToken'
+            );
+          });
+      });
+    });
+
+    Expect.describe('when features.skipIdpFactorVerificationBtn is false', function() {
+      let settings = {};
+
+      beforeEach(() => {
+        settings = { 'features.skipIdpFactorVerificationBtn': false };
+      });
+
+      itp('shows the right subtitle', function() {
+        return setup(settings)
+          .then(function(test) {
+            expectSubtitleToBe(test, 'Clicking below will redirect to verification with IDP factor');
+          });
+      });
+
+      itp('shows the button bar', function() {
+        return setup(settings)
+          .then(function(test) {
+            expect(test.form.buttonBar().hasClass('hide')).toBe(false);
+          });
+      });
+
+      itp('hides the waiting spinner', function() {
+        return setup(settings)
+          .then(function(test) {
+            expect(test.form.hasSpinner()).toBe(false);
+          });
+      });
+
+      itp('redirects to third party when Verify button is clicked', function() {
+        spyOn(SharedUtil, 'redirect');
+        return setup(settings)
+          .then(function(test) {
+            test.setNextResponse([resChallengeClaimsProvider, resSuccess]);
+            test.form.submit();
+            return Expect.waitForSpyCall(SharedUtil.redirect);
+          })
+          .then(function() {
+            expect(SharedUtil.redirect).toHaveBeenCalledWith(
+              'http://rain.okta1.com:1802/sso/idps/idpId?stateToken=testStateToken'
+            );
+          });
+      });
+
+      itp('does not redirect to third party automatically', function() {
+        jest.spyOn(SharedUtil, 'redirect');
+        return setup(settings)
+          .then(function(test) {
+            test.setNextResponse([resChallengeClaimsProvider, resSuccess]);
+            test.form.initialize();
```

Comment:
as mentioned in a previous comment, we shouldn't be calling `initialize` explicitly, that function is called when the controller is rendered.